15-connect-app-and-store/README.md

```diff
@@ -0,0 +1,527 @@
+
+A component which has been passed to connect takes on a different term: a [container](http://redux.js.org/docs/basics/UsageWithReact.html#presentational-and-container-components). The term comes from an abstraction suggested in redux documentation that components should be split into two basic groups: those which handle *presentation* and those which handle *data*. "Containers" are passed to `connect()` and so are aware of the state, and are able to dispatch actions, while presentational components simply consume their props and maintain state for UI purposes.
+
+In this step we are going to do all that is left in order to connect our React application to our Redux store and actions. This will include:
+* Creating our `store` in `App.js`
+* Applying `redux-thunk` as a middleware to handle async actions
+* Making `<Provider />` the top level component
+* Refactoring of `<App />` such that its primary job is creating the store, and hydrating the app. Moving the JSX into a new container component: `<Page />`
+* `connect()` `<Page />` so that we can `dispatch()` actions, and access our App's state
+
+The [tasks](#tasks) and [exercises](#exercises) below will address each of the steps above and when we finish, will have a fully functioning Redux app.
+
+As always, if you run into trouble with the [tasks](#tasks) or [exercises](#exercises), you can take a peek at the final [source code](src/).
+
+## Tasks
+
+First things first, lets install the libraries necessary to make this happen. If you are starting the workshop from this step, or have already run `yarn install` in this directory please disregard. For everyone else, run:
+```
+yarn add redux
+yarn add react-redux
+yarn add redux-thunk
+```
+
+Once that finishes we are good to go!
+
+Lets create a new folder [`containers/`](src/containers/) and within that folder add a new file [`Page.js`](src/containers/Page.js). Lets quickly add `<Page />` to export.
+
+```js
+import React, {PureComponent} from 'react';
+
+export default class Page extends PureComponent {
+  render() {
+    return (
+      <div />
+    )
+  }
+}
+```
+
+Next we are going to do is some ground work to set up the modified app structure. We want the `<Page />` to deal with:
+* the layout
+* handling user interactions
+* UI state
+
+The `<App />` should concern itself with:
+* setting up the `store`
+* initial hydration of the App
+
+So, first pull out the **JSX**, **helper components**, **component imports**, **event handlers** (such as `_handleItemSelect`), and **state** from `App.js`. Move them into `Page.js`.
```

Comment:
I wonder if it would've been easier to just copy `src/App.js` to `src/containers/Page.js` and then rename the component within? It'd be nice to avoid pasting the long block of code below and only the necessary changes to type